lib/include/jxl/encode.h

```diff
@@ -41,8 +41,10 @@ JXL_EXPORT uint32_t JxlEncoderVersion(void);
 typedef struct JxlEncoderStruct JxlEncoder;
 
 /**
- * Opaque structure that holds frame specific encoding options for a JPEG XL
- * encoder.
+ * The options for a single image frame to be encoded with the JXL encoder.
+ * This includes options such as compression speed, as well as frame metadata.
+ *
+ * Despite the name JxlEncoderOptions, it represents options of a frame instead.
```

Comment:
Maybe it would make sense to rename it (to `JxlEncoderFrameSettings` or something) and for now keep the old name as a deprecated synonym (that maybe we can get rid of when we finalize the api).

I think "EncoderOptions" is misleading if it doesn't just include encode options (like effort and quality target) but also actual frame data (like duration and crop dimensions/offset). The word "Settings" is similar to "Options" but implies that it can also be data, not just optional options. 

So I would use `frame_settings` instead of `frame_options` throughout, and also rename this struct to `JxlEncoderFrameSettings` with for now a deprecated synonym `JxlEncoderOptions` in order to not break existing encoders.

Reply:
Another option is to keep JxlEncoderOptions as-is (though better documented), and not use them for the frame metadata purpose. But then have a JxlEncoderSetFrameHeader that takes an encoder object, instead of a JxlEncoderOptions

But it would then be odd that JxlEncoderAddImageFrame is done with a JxlEncoderOptions, while JxlEncoderSetFrameHeader that sets the header of that frame would be done with JxlEncoder instead.

About renaming: that would break backwards compatibility of course.

Reply:
The frame_settings rename was done on another MR, this comment is obsolete